Adds premium interaction docs (#6875)
hammerandchisel/discord-api-docs: docs/interactions/Message_Components.md

| type       | integer                                             | `2` for a button                                                                                                                                     |
| style      | integer                                             | A [button style](#DOCS_INTERACTIONS_MESSAGE_COMPONENTS/button-object-button-styles)                                                                  |
| label?     | string                                              | Text that appears on the button; max 80 characters                                                                                                   |
| emoji?     | partial [emoji](#DOCS_RESOURCES_EMOJI/emoji-object) | `name`, `id`, and `animated`                                                                                                                         |
| custom_id? | string                                              | Developer-defined identifier for the button; max 100 characters                                                                                      |

> | sku_id?    | snowflake                                           | Identifier for a purchasable SKU, only available when using [ButtonStyle.Premium](#DOCS_INTERACTIONS_MESSAGE_COMPONENTS/button-object-button-styles) |

Reviewer: ```suggestion
| sku_id?    | snowflake                                           | Identifier for a purchasable [SKU](#DOC_MONETIZATION_SKUS/sku-object), only available when using premium-style buttons                    |
```

| label?     | string                                              | Text that appears on the button; max 80 characters                                  |
| emoji?     | partial [emoji](#DOCS_RESOURCES_EMOJI/emoji-object) | `name`, `id`, and `animated`                                                        |
| custom_id? | string                                              | Developer-defined identifier for the button; max 100 characters                     |

> | sku_id?    | string                                              | Identifier for a purchasable SKU, only available when using [ButtonStyle.Premium](#DOCS_INTERACTIONS_MESSAGE_COMPONENTS/button-object-button-styles) |

Reviewer 2: ```suggestion
| sku_id?    | snowflake                                           | Identifier for a purchasable SKU, only available when using [ButtonStyle.Premium](#DOCS_INTERACTIONS_MESSAGE_COMPONENTS/button-object-button-styles) |
```

Author: it is indeed a snowflake!

| Success   | 3     | green                    | `custom_id`    |
| Danger    | 4     | red                      | `custom_id`    |
| Link      | 5     | grey, navigates to a URL | `url`          |

> | Premium   | 6     | gradient                 | `sku_id`       |

Reviewer 3: RIP the gradient :(
```suggestion
| Premium   | 6     | blurple                  | `sku_id`       |
```

Author: RIP the gradient.

![An image showing the different button styles](button-styles.png)

When a user clicks on a button, your app will receive an [interaction](#DOCS_INTERACTIONS_RECEIVING_AND_RESPONDING/interaction-object) including the message the button was on:

### Button Design Guidelines

###### General Button copy

> - 34 characters max with icon or emoji
> - 38 characters max without icon or emoji

Reviewer 4: Consistency: All the other list items end with a dot.
```suggestion
- 34 characters max with icon or emoji.
- 38 characters max without icon or emoji.
```
